### PR TITLE
Selenium: Rework the 'GitCompareTest' by use the 'kohsuke Github' library

### PR DESCRIPTION
**Pull request by artaleks9**
### What does this PR do?
* Rework the _GitCompareTest_ by use the _kohsuke Github_ library
* Split the selenium test _GitCompareTest_ on the three test methods
* Rework the methods in the _GitCompare_ page object using the _SeleniumWebDriverHelper_ and _WebDriverWaitFactory_

### What issues does this PR fix or reference?
#9459, #9516

### CODE REVIEW

**Comment by artaleks9** [comment]
ci-test

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitCompareTest.java [hunk 1]
```diff
@@ -65,181 +94,152 @@
...
     git.waitGitCompareFormIsOpen();
-    git.waitExpTextIntoCompareLeftEditor("public class A");
-    git.setFocusOnLeftGitCompareEditor();
-    git.setCursorToLine(2, LEFT_COMPARE_ST);
-    git.typeTextIntoGitCompareEditor("//***che***codenvy***");
-    git.waitExpTextIntoCompareLeftEditor("//***che***codenvy***");
-    git.clickOnGitCompareCloseButton();
-    askDialog.confirmAndWaitClosed();
-    git.waitGitCompareFormIsClosed();
+    git.waitExpTextIntoCompareLeftEditor("public class NewClass");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitCompareTest.java:146]: Could you please add comment into the code with description of steps? For example - to explain where we are checking compare after adding, and where - after deleting.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitCompareTest.java [hunk 2]
```diff
@@ -65,181 +94,152 @@
...
+    // check the 'git compare' for another local branch
+    menu.runCommand(GIT, COMPARE_TOP, COMPARE_WITH_BRANCH);
     git.waitGitCompareBranchFormIsOpen();
     git.selectBranchIntoGitCompareBranchForm("newbranch");
     git.clickOnCompareBranchFormButton();
-    loader.waitOnClosed();
+    git.waitGroupGitCompareIsOpen();
+    git.waitExpTextInGroupGitCompare(TEXT_GROUP);
+
+    // check the 'Next' and 'Previous' diff buttons
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitCompareTest.java]: Could you, please, divide the code which check 'Next' button, from those which check 'Previous' button?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitCompare.java [hunk 3]
```diff
@@ -56,6 +64,8 @@ public GitCompare(
     String LEFT_COMPARE_STATUS =
         "//div[@id='gwt-debug-compareParentDiv_left_status_id' and text()='%s']";
     String RIGHT_COMPARE_EDITOR = "gwt-debug-compareParentDiv_right_editor_id";
+    String NEXT_DIFF_BUTTON = "git-compare-next-diff-btn";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitCompare.java]: Is it name or id? It's better to express nature of locator in the constant name by adding the prefix `_ID` or `_NAME`

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitCompare.java [hunk 4]
```diff
@@ -129,10 +154,29 @@ public String getTextFromRightEditorCompare() {
...
+    seleniumWebDriverHelper.waitTextPresence(leftCompareEditor, expText);
+  }
+
+  /**
+   * wait expected text is not present into left editor of the 'Git Compare'
+   *
+   * @param expText expected value
+   */
+  public void waitTextNotPresentIntoLeftEditor(String expText) {
+    webDriverWaitFactory
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitCompare.java]: It's better to rewrite the code to use SeleniumWebDriverHelper class.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitCompare.java [hunk 5]
```diff
@@ -141,15 +185,16 @@ public void waitExpectedTextIntoLeftEditor(String expText) {
    * @param expText expected value
    */
   public void waitTextNotPresentIntoRightEditor(String expText) {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
+    webDriverWaitFactory
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitCompare.java]: It's better to rewrite the code to use SeleniumWebDriverHelper class.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 6]
```diff
@@ -674,6 +674,37 @@ public void waitTextPresence(By elementLocator, String expectedText) {
...
+   * Waits during {@code timeout} until specified {@code element} does not contain the specified
+   * {@code expectedText}.
+   *
+   * <p>Note! Text is extracted by {@link WebElement#getText()} method.
+   *
+   * @param element element which should be checked
+   * @param expectedText text which should be not presented
+   * @param timeout waiting time in seconds
+   */
+  public void waitTextIsNotPresence(WebElement element, String expectedText, int timeout) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java]: `waitTextIsNotPresence` > `waitTextIsNotPresented`

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 7]
```diff
@@ -674,6 +674,37 @@ public void waitTextPresence(By elementLocator, String expectedText) {
...
+  /**
+   * Waits until {@link WebElement} which defined by {@code element} does not contain the specified
+   * {@code expectedText}.
+   *
+   * <p>Note! Text is extracted by {@link WebElement#getText()} method.
+   *
+   * @param element element which should be checked
+   * @param expectedText text which should be not presented
+   */
+  public void waitTextIsNotPresence(WebElement element, String expectedText) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java]: `waitTextIsNotPresence` > `waitTextIsNotPresented`

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 8]
```diff
@@ -674,6 +674,37 @@ public void waitTextPresence(By elementLocator, String expectedText) {
...
+   * @param element element which should be checked
+   * @param expectedText text which should be not presented
+   * @param timeout waiting time in seconds
+   */
+  public void waitTextIsNotPresence(WebElement element, String expectedText, int timeout) {
+    webDriverWaitFactory
+        .get(timeout)
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> !(waitVisibility(element).getText().contains(expectedText)));
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java]: driver -> !(waitVisibilityAndGetText(element).contains(expectedText)));

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 9]
```diff
@@ -674,6 +674,37 @@ public void waitTextPresence(By elementLocator, String expectedText) {
...
+  }
+
+  /**
+   * Waits until {@link WebElement} which defined by {@code element} does not contain the specified
+   * {@code expectedText}.
+   *
+   * <p>Note! Text is extracted by {@link WebElement#getText()} method.
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java:699]: Also, you should create an override for a case where text extracted by "getAttribute". Also with argument timeout and default timeout.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 10]
```diff
@@ -674,6 +674,37 @@ public void waitTextPresence(By elementLocator, String expectedText) {
...
+   *
+   * @param element element which should be checked
+   * @param expectedText text which should be not presented
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java]: `should be not` > `should not be`

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 11]
```diff
@@ -674,6 +674,37 @@ public void waitTextPresence(By elementLocator, String expectedText) {
...
   }
 
+  /**
+   * Waits during {@code timeout} until specified {@code element} does not contain the specified
+   * {@code expectedText}.
+   *
+   * <p>Note! Text is extracted by {@link WebElement#getText()} method.
+   *
+   * @param element element which should be checked
+   * @param expectedText text which should be not presented
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java]: `should be not` > `should not be`

**Review by Ohrimenko1988 — APPROVED** [review]

**Review by dmytro-ndp — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
ci-build

**Pull request merged**